### PR TITLE
Migrate prof dump

### PR DESCRIPTION
**Pull request by yinan1048576**
Replace the original `prof_dump_write` by the new buffered writer.

### CODE REVIEW

**Comment by yinan1048576** [comment]
Two remarks:

- The new buffered writer is supposed to be encapsulated in the sense that the outside world should not directly alter any internal data i.e. fields in a `buf_writer_arg_t` variable. However I found `prof_dump_maps` may have to be an exception: it's using the buffer to "pump" contents directly from an input file to an output file, which seems hard to be generalized, so I'm letting it directly touch internal fields of the `buf_writer_arg_t` variable. This is the only place I know of doing buffered file piping, but if it turns out that it also appears in other places, I'll think of ways to generalize it out to e.g. another API in the buffered writer.

- I removed the `propagate_err` parameter in `prof_dump_close`, but it seems also removable in many (or probably all?) other places, since the static `buf_writer_arg_t` variable already carries it and it never changes throughout the entire `prof_dump_file` call. I figured that this could be a change of a relatively larger scale, so I'm trying not to pollute this pull request for now.

**Comment by yinan1048576** [comment]
Rebase.

File: src/prof.c [hunk 1]
```diff
@@ -1257,70 +1265,37 @@ prof_dump_flush(bool propagate_err) {
...
 	}
-	prof_dump_buf_end = 0;
-
-	return ret;
 }
 
-static bool
-prof_dump_close(bool propagate_err) {
-	bool ret;
+static buf_writer_arg_t prof_dump_buf_arg =
```
Comment (interwq) [inline src/prof.c]: Is this covered by any lock?

**Comment by interwq** [comment]
Let's continue this @yinan1048576 ?

**Comment by yinan1048576** [comment]
Yes, my plan is to resume after #1589 is in.

**Comment by yinan1048576** [comment]
Becomes part of #1795 and #1794.

**Pull request closed without merging**